src/harness/reference_models/aggregate_interference/aggregate_interference.py

```diff
@@ -0,0 +1,362 @@
+  aggregate_interference = aggregateInterferenceForPoint(protection_point, 
+                             protection_channels, interf.ESC_CAT_B_LOW_FREQ_HZ, 
+                             interf.ESC_CAT_B_HIGH_FREQ_HZ, grant_list, None, 
+                             esc_antenna_info, interf.ProtectedEntityType.ESC_CAT_B)
+
+  return aggregate_interference
+
+
+def calculateAggregateInterferenceForGwpz(gwpz_record, cbsd_list):
+  """Calculates aggregate interference for GWPZ.
+  
+  Calculates aggregate interference for each protection constraint in the GWPZ
+  based on the interferences caused by CBSD's grants in the neighborhood of
+  the GWPZ protection entity.
+
+  Args:
+    gwpz_record: GWPZ protection entity
+    cbsd_list: list of CBSD objects containing RegistrationRequests and GrantRequests
+  Returns:
+    Aggregate interference to GWPZ in the dictionary format,where the key is a tuple. 
+    {(latitude, longitude, low_frequency, high_frequency,entity_type):
+    aggregate_interference(dBm)}
+  """
+
+  grant_list = interf.getAllGrantInformationFromCbsdDataDump(cbsd_list)
+  gwpz_region = gwpz_record['zone']['features'][0]['properties']['gwpzRegionType']
+
+  # Get Fine Grid Points for a GWPZ protection area
+  protection_points = utils.GridPolygon(gwpz_record)
+  gwpz_freq_range = gwpz_record['deploymentParam']['operationParam']['operationFrequencyRange']
+  gwpz_low_freq = gwpz_freq_range['lowFrequency']
+  gwpz_high_freq = gwpz_freq_range['highFrequency']
+
+  # Get channels over which area incumbent needs partial/full protection
+  protection_channels = interf.getProtectedChannels(gwpz_low_freq, gwpz_high_freq)
+  for protection_point in protection_points:
+    aggregate_interference = aggregateInterferenceForPoint(protection_point, 
+                               protection_channels, gwpz_low_freq, 
+                               gwpz_high_freq, grant_list, None, None, 
+                               interf.ProtectedEntityType.GWPZ_AREA, 
+                               gwpz_region)
+  return aggregate_interference
+
+
+def calculateAggregateInterferenceForPpa(ppa_record, pal_list, cbsd_list):
+  """Calculates aggregate interference for PPA.
+  
+  Calculates aggregate interference for each protection constraint in the PPA
+  based on the interferences caused by CBSD's grants in the neighborhood of
+  the PPA protection entity.
+
+  Args:
+    ppa_record: PPA protection entity
+    pal_list: list of PAL records
+    cbsd_list: list of CBSD objects containing RegistrationRequests and GrantRequests
+  Returns:
+    Aggregate interference to PPA in the dictionary format,where the key is a tuple. 
+    {(latitude, longitude, low_frequency, high_frequency,entity_type):
+    aggregate_interference(dBm)}
+  """
+  grant_list = interf.getAllGrantInformationFromCbsdDataDump(cbsd_list)
+
+  # Get Fine Grid Points for a PPA protection area
+  protection_points = utils.GridPolygon(ppa_record)
```

Comment:
Same: you need to pass the resolution of gridding

Reply:
Updated   resolution in arcsecs to 2